src/host/thread_host.hpp

```diff
@@ -106,6 +106,56 @@ enum ThreadEnabledState
+constexpr Error kErrorFailed                     = OT_ERROR_FAILED;
+constexpr Error kErrorDrop                       = OT_ERROR_DROP;
+constexpr Error kErrorNoBufs                     = OT_ERROR_NO_BUFS;
+constexpr Error kErrorNoRoute                    = OT_ERROR_NO_ROUTE;
+constexpr Error kErrorBusy                       = OT_ERROR_BUSY;
+constexpr Error kErrorParse                      = OT_ERROR_PARSE;
+constexpr Error kErrorInvalidArgs                = OT_ERROR_INVALID_ARGS;
+constexpr Error kErrorSecurity                   = OT_ERROR_SECURITY;
+constexpr Error kErrorAddressQuery               = OT_ERROR_ADDRESS_QUERY;
+constexpr Error kErrorNoAddress                  = OT_ERROR_NO_ADDRESS;
+constexpr Error kErrorAbort                      = OT_ERROR_ABORT;
+constexpr Error kErrorNotImplemented             = OT_ERROR_NOT_IMPLEMENTED;
+constexpr Error kErrorInvalidState               = OT_ERROR_INVALID_STATE;
+constexpr Error kErrorNoAck                      = OT_ERROR_NO_ACK;
+constexpr Error kErrorChannelAccessFailure       = OT_ERROR_CHANNEL_ACCESS_FAILURE;
+constexpr Error kErrorDetached                   = OT_ERROR_DETACHED;
+constexpr Error kErrorFcs                        = OT_ERROR_FCS;
+constexpr Error kErrorNoFrameReceived            = OT_ERROR_NO_FRAME_RECEIVED;
+constexpr Error kErrorUnknownNeighbor            = OT_ERROR_UNKNOWN_NEIGHBOR;
+constexpr Error kErrorInvalidSourceAddress       = OT_ERROR_INVALID_SOURCE_ADDRESS;
+constexpr Error kErrorAddressFiltered            = OT_ERROR_ADDRESS_FILTERED;
+constexpr Error kErrorDestinationAddressFiltered = OT_ERROR_DESTINATION_ADDRESS_FILTERED;
+constexpr Error kErrorNotFound                   = OT_ERROR_NOT_FOUND;
+constexpr Error kErrorAlready                    = OT_ERROR_ALREADY;
+constexpr Error kErrorIp6AddressCreationFailure  = OT_ERROR_IP6_ADDRESS_CREATION_FAILURE;
+constexpr Error kErrorNotCapable                 = OT_ERROR_NOT_CAPABLE;
+constexpr Error kErrorResponseTimeout            = OT_ERROR_RESPONSE_TIMEOUT;
+constexpr Error kErrorDuplicated                 = OT_ERROR_DUPLICATED;
+constexpr Error kErrorReassemblyTimeout          = OT_ERROR_REASSEMBLY_TIMEOUT;
+constexpr Error kErrorNotTmf                     = OT_ERROR_NOT_TMF;
+constexpr Error kErrorNotLowpanDataFrame         = OT_ERROR_NOT_LOWPAN_DATA_FRAME;
+constexpr Error kErrorLinkMarginLow              = OT_ERROR_LINK_MARGIN_LOW;
+constexpr Error kErrorInvalidCommand             = OT_ERROR_INVALID_COMMAND;
+constexpr Error kErrorPending                    = OT_ERROR_PENDING;
+constexpr Error kErrorRejected                   = OT_ERROR_REJECTED;
+constexpr Error kErrorGeneric                    = OT_ERROR_GENERIC;
+
+// Proprietary errors: -32768 ~ -1
+constexpr Error kErrorUnsupportedChannel = -1; // The channel provided is not supported.
+constexpr Error kErrorDisabled           = -2; // The action didn't succeed because Thread is in Disabled state.
```

Comment:
Is it possible to use `kInvalidState`?

Reply:
`ERROR_THREAD_DISABLED` indicates that the operation failed because the Thread network feature is disabled. The caller should not retry before Thread lis enabled.

Reply:
Then, perhaps we should use `kErrorNotImplemented`?

Reply:
This error specifically indicates that the caller should call the `ThreadHost::SetThreadEnabled` first. In Android, it indicates the caller to call [SetThreadEnabled API](https://cs.android.com/android/platform/superproject/main/+/main:out/soong/.intermediates/external/ot-br-posix/src/android/aidl/ot-daemon-aidl-ndk-source/gen/include/aidl/com/android/server/thread/openthread/IOtDaemon.h;l=106;drc=a94b8b985b7bdf228dea2588a1302a29ccc2443a;bpv=0;bpt=1). So it doesn't mean that the code, the Thread feature is not statically enabled.

Reply:
Can we have a mapping between go/canonical-codes and Thread's error codes? It looks like the FAILED_PRECONDITION should be mapped to kInvalidState from the description of both error codes.